### PR TITLE
Differentiate translations

### PR DESCRIPTION
**Pull request by sisskind**

### CODE REVIEW

**Comment by brianhatchl** [comment]
Getting a js error.  I don't have any custom translations.
>Translation.js:52 Uncaught TypeError: Cannot read property 'DEFAULT' of undefined
    at HTMLAnchorElement.<anonymous> (Translation.js:52)
    at HTMLAnchorElement.arguments.length.each.value.function.value.textContent (d3.v3.js:1021)
    at d3.v3.js:1254
    at d3_selection_each (d3.v3.js:1261)
    at Array.d3_selectionPrototype.each (d3.v3.js:1253)
    at Array.d3_selectionPrototype.text (d3.v3.js:1020)
    at Translation.js:51
    at rest.js:606
    at Object.<anonymous> (d3.v3.js:2172)
    at Object.event (d3.v3.js:584)

**Comment by rmrice** [comment]
Same as @brianhatchl. Also Import Single Dataset seems to be broken:

screenshot from 2017-09-01 11-39-52

**Comment by sisskind** [comment]
@brianhatchl @rmrice just slammed in code that should fix that.

**Comment by brianhatchl** [comment]
I think d is null.  Still getting error.  My UI shows a blank translation at the top.  
screenshot from 2017-09-01 13-45-57

But the server response seems good:
```
[
  {
    "PATH": "translations/GeoNames.js",
    "CANEXPORT": false,
    "DESCRIPTION": "GeoNames.org",
    "DEFAULT": true,
    "NAME": "GEONAMES"
  },
  {
    "PATH": "translations/GGDMv30.js",
    "CANEXPORT": true,
    "DESCRIPTION": "Ground-Warfighter Geospatial Data Model (GGDM) v3.0",
    "DEFAULT": true,
    "NAME": "GGDMv30"
  },
  {
    "PATH": "translations/GIFD_D4.js",
    "CANEXPORT": false,
    "DESCRIPTION": "Geospatial Intelligence Feature Database (GIFD) D4",
    "DEFAULT": true,
    "NAME": "GIFD D4"
  },
  {
    "PATH": "translations/MGCP_TRD4.js",
    "CANEXPORT": true,
    "DESCRIPTION": "Multinational Geospatial Co-production Program (MGCP) TRD3&4",
    "DEFAULT": true,
    "NAME": "MGCP"
  },
  {
    "CANEXPORT": true,
    "DESCRIPTION": "OpenStreetMap.org (OSM)",
    "EXPORTPATH": "translations/RenderDb.js",
    "IMPORTPATH": "translations/OSM_Ingest.js",
    "DEFAULT": true,
    "NAME": "OSM"
  },
  {
    "PATH": "translations/TDSv40.js",
    "CANEXPORT": true,
    "DESCRIPTION": "NSG Topographic Data Store (TDS) v4.0",
    "DEFAULT": true,
    "NAME": "TDSv40"
  },
  {
    "PATH": "translations/TDSv61.js",
    "CANEXPORT": true,
    "DESCRIPTION": "NSG Topographic Data Store (TDS) v6.1",
    "DEFAULT": true,
    "NAME": "TDSv61"
  }
]
```

**Comment by sisskind** [comment]
Not sure why this is working on mine then.  Will keep digging.

**Comment by sisskind** [comment]
@brianhatchl are you using localhost:8080 or localhost:8888?

**Comment by brianhatchl** [comment]
8080 from hoot-server

**Comment by sisskind** [comment]
Do you have any custom translations?

**Comment by sisskind** [comment]
Found the problem -- its if you don't have any custom translations. Stand by.

**Comment by sisskind** [comment]
@brianhatchl @rmrice just slammed in code that should fix that.

File: js/hoot/model/rest.js [hunk 1]
```diff
@@ -585,6 +585,27 @@ Hoot.model.REST = function (command, data, callback, option) {
...
+                    resp = dTrans.true.concat(dTrans.undefined);
+                } else {
+                    resp = dTrans.true;
+                }
+
+
+            } catch (eTrans) {
+                resp.sort(function(x,y){ return (x.DEFAULT === true) ? 0 : x.DEFAULT ? -1 : 1; });
+            }
+
```
Comment (brianhatchl) [inline js/hoot/model/rest.js]: Seems to work, but this code is hella confusing.

The purpose is to sort by DEFAULT = true, then false, then undefined and alphabetically within those?
Could we just use a sort function that utilized both rules?

I'll try to create a proposed sort function and mocha test for this.

**Comment by sisskind** [comment]
@brianhatchl I had something similar to what you have but wouldn't alpha first one especially if custom.

**Comment by sisskind** [comment]
@brianhatchl just put it through the paces and its working nicely.  Thanks for taking the time to review the code.

**Comment by brianhatchl** [comment]
run :cucumber:

**Comment by rmrice** [comment]
run 🥒

**Comment by sisskind** [comment]
run :cucumber:

**Comment by rmrice** [comment]
run :cucumber:

**Pull request closed without merging**